### PR TITLE
[#34] Bump externals for 4-2-stable

### PR DESCRIPTION
**Pull request by alanking**
Builds successfully on ubuntu:16.04 and centos:7 images after installing git (for cloning), sudo, python, and curl. Still running tests with iRODS server and plugins built against these externals.

### CODE REVIEW

File: build.py [hunk 1]
```diff
@@ -267,6 +267,11 @@ def build_package(target):
             if target == 'boost':
                 # using boostorg namespace instead of forking 50+ relatively linked submodules
                 run_cmd(['git', 'clone', 'https://github.com/boostorg/boost'])
+            elif target == 'avro':
+                run_cmd(['git', 'clone', 'https://github.com/{0}/{1}'.format('apache', target)])
+            # TODO: Remove me
+            elif target == 'mungefs':
```
Comment (alanking) [inline build.py]: Need to remove this upon merging of irods/mungefs#15.

Comment (trel) [reply build.py]: merged

File: build.py [hunk 2]
```diff
@@ -267,6 +267,8 @@ def build_package(target):
             if target == 'boost':
                 # using boostorg namespace instead of forking 50+ relatively linked submodules
                 run_cmd(['git', 'clone', 'https://github.com/boostorg/boost'])
+            elif target == 'avro':
```
Comment (trel) [inline build.py]: we have https://github.com/irods/avro already?

File: versions.json [hunk 3]
```diff
@@ -42,7 +42,7 @@
         "externals_root": "opt/irods-externals",
         "build_steps": [
             "mkdir -p build",
-            "cd build; TEMPLATE_CMAKE_EXECUTABLE -G 'Unix Makefiles' -DENABLE_TESTING=OFF -DCMAKE_INSTALL_PREFIX=TEMPLATE_INSTALL_PREFIX -DCMAKE_CXX_COMPILER=TEMPLATE_CLANGPP_EXECUTABLE -DCMAKE_SHARED_LINKER_FLAGS='-LTEMPLATE_CLANG_CPP_LIBRARIES -stdlib=libc++' -DCMAKE_EXE_LINKER_FLAGS='-LTEMPLATE_CLANG_CPP_LIBRARIES -stdlib=libc++' -DCMAKE_INCLUDE_DIRECTORIES_BEFORE=ON -DCMAKE_CXX_FLAGS='-ITEMPLATE_CLANG_CPP_HEADERS' -DBUILD_ONLY='s3;sts' ..",
+            "cd build; TEMPLATE_CMAKE_EXECUTABLE -G 'Unix Makefiles' -DENABLE_TESTING=OFF -DCMAKE_INSTALL_PREFIX=TEMPLATE_INSTALL_PREFIX -DCMAKE_CXX_COMPILER=TEMPLATE_CLANG_EXECUTABLE -DCMAKE_SHARED_LINKER_FLAGS='-LTEMPLATE_CLANG_CPP_LIBRARIES -stdlib=libc++' -DCMAKE_EXE_LINKER_FLAGS='-LTEMPLATE_CLANG_CPP_LIBRARIES -stdlib=libc++' -DCMAKE_INCLUDE_DIRECTORIES_BEFORE=ON -DCMAKE_CXX_FLAGS='-ITEMPLATE_CLANG_CPP_HEADERS -nostdinc++' -DBUILD_ONLY='s3;sts' ..",
```
Comment (trel) [inline versions.json]: i think we still want the `TEMPLATE_CLANGPP_EXECUTABLE`?

File: versions.json [hunk 4]
```diff
@@ -266,7 +266,7 @@
         "fpm_directories": ["bin","include","lib"]
     },
     "mungefs": {
-        "commitish": "1.0.2",
+        "commitish": "bump",
```
Comment (trel) [inline versions.json]: tweak

File: versions.json (hunk 3 above)
Comment (alanking) [reply versions.json]: Yes, you're right. I think this diff came from master - will replace. Interestingly, it seems to still build fine? I guess that remains to be seen through S3 plugin testing against this external.

File: versions.json (hunk 4 above)
Comment (alanking) [reply versions.json]: Using 1.0.3 which will be released shortly.

**Pull request merged**